File: 30 days of code/Day 08 - Dictionaries and maps/Dictionaries and maps.cs
/**
 *              DICTIONARIES AND MAPS | C#
 * Domain       30 DAYS OF CODE
 * Sub domain   DAY 08
 */

using System;
using System.Collections.Generic;
using System.IO;

class Program
{
    //static TextReader read = Console.In;
    static StreamReader read = new StreamReader("input.txt");

    static void Main(string[] args)
    {
        int n = int.Parse(read.ReadLine());
        Dictionary<string, int> phoneBook = new Dictionary<string, int>(n);
        for (int i = 0; i < n; ++i)
        {
            string[] line = read.ReadLine().Split(' ');
            phoneBook.Add(line[0], int.Parse(line[1]));
        }

        string s = null;
        while ((s = read.ReadLine()) != null)
        {
            if (phoneBook.ContainsKey(s))
            {
                Console.WriteLine($"{s}={phoneBook[s]}");
            }
            else
            {
                Console.WriteLine("Not found");
            }
        }

        read.Close();
    }
}

File: 30 days of code/Day 08 - Dictionaries and maps/Dictionaries and maps.cpp
/**
*              DICTIONARIES AND MAPS | C++
* Domain       30 DAYS OF CODE
* Sub domain   DAY 08
*/

#include "stdafx.h"
#include <iostream>
#include <fstream>
#include <map>
#include <string>

using namespace std;

ifstream file("input.txt");
istream& f = file;

void main() {
	int n; f >> n;
	map<string, int> phoneBook;
	for (int i = 0; i < n; ++i) {
		string name; f >> name;
		int phone; f >> phone;
		phoneBook[name] = phone;
	}

	string s;
	while (f >> s) {
		if (phoneBook.find(s) != phoneBook.end()) {
			cout << s << "=" << phoneBook.find(s)->second << endl;
		}
		else {
			cout << "Not found" << endl;
		}
	}
}